### ISSUE TITLE
Support for check_mode in all *_info modules

### ISSUE THREAD

**Issue by Klaas-** [ansible/ansible_collections_azure#490]
##### SUMMARY
All _info modules I tested do not support checkmode
`{"changed": false, "msg": "remote module (azure.azcollection.azure_rm_publicipaddress_info) does not support check mode"}`

List of all modules I manually tested:
* azure_rm_adserviceprincipal_info
* azure_rm_loganalyticsworkspace_info
* azure_rm_publicipaddress_info
* azure_rm_resourcegroup_info

These modules by nature only retrieve information, so it should always be safe to run them in check mode, right? :)

Workaround:
you can explicitly tell them to run even when in check mode
`check_mode: no`

I also checked the sources and I made a list of all the ***_info** modules that **do not** support check mode:
```
./plugins/modules/azure_rm_adpassword_info.py
./plugins/modules/azure_rm_adserviceprincipal_info.py
./plugins/modules/azure_rm_aksupgrade_info.py
./plugins/modules/azure_rm_aksversion_info.py
./plugins/modules/azure_rm_aks_info.py
./plugins/modules/azure_rm_apimanagementservice_info.py
./plugins/modules/azure_rm_applicationsecuritygroup_info.py
./plugins/modules/azure_rm_appserviceplan_info.py
./plugins/modules/azure_rm_automationaccount_info.py
./plugins/modules/azure_rm_autoscale_info.py
./plugins/modules/azure_rm_availabilityset_info.py
./plugins/modules/azure_rm_azurefirewall_info.py
./plugins/modules/azure_rm_backuppolicy_info.py
./plugins/modules/azure_rm_cdnendpoint_info.py
./plugins/modules/azure_rm_cdnprofile_info.py
./plugins/modules/azure_rm_cognitivesearch_info.py
./plugins/modules/azure_rm_containerinstance_info.py
./plugins/modules/azure_rm_containerregistryreplication_info.py
./plugins/modules/azure_rm_containerregistrywebhook_info.py
./plugins/modules/azure_rm_containerregistry_info.py
./plugins/modules/azure_rm_cosmosdbaccount_info.py
./plugins/modules/azure_rm_datalakestore_info.py
./plugins/modules/azure_rm_deployment_info.py
./plugins/modules/azure_rm_devtestlabarmtemplate_info.py
./plugins/modules/azure_rm_devtestlabartifactsource_info.py
./plugins/modules/azure_rm_devtestlabartifact_info.py
./plugins/modules/azure_rm_devtestlabcustomimage_info.py
./plugins/modules/azure_rm_devtestlabenvironment_info.py
./plugins/modules/azure_rm_devtestlabpolicy_info.py
./plugins/modules/azure_rm_devtestlabschedule_info.py
./plugins/modules/azure_rm_devtestlabvirtualmachine_info.py
./plugins/modules/azure_rm_devtestlabvirtualnetwork_info.py
./plugins/modules/azure_rm_devtestlab_info.py
./plugins/modules/azure_rm_dnsrecordset_info.py
./plugins/modules/azure_rm_dnszone_info.py
./plugins/modules/azure_rm_functionapp_info.py
./plugins/modules/azure_rm_galleryimageversion_info.py
./plugins/modules/azure_rm_galleryimage_info.py
./plugins/modules/azure_rm_gallery_info.py
./plugins/modules/azure_rm_hdinsightcluster_info.py
./plugins/modules/azure_rm_image_info.py
./plugins/modules/azure_rm_iothub_info.py
./plugins/modules/azure_rm_keyvaultkey_info.py
./plugins/modules/azure_rm_keyvaultsecret_info.py
./plugins/modules/azure_rm_keyvault_info.py
./plugins/modules/azure_rm_loadbalancer_info.py
./plugins/modules/azure_rm_lock_info.py
./plugins/modules/azure_rm_loganalyticsworkspace_info.py
./plugins/modules/azure_rm_managementgroup_info.py
./plugins/modules/azure_rm_mariadbconfiguration_info.py
./plugins/modules/azure_rm_mariadbdatabase_info.py
./plugins/modules/azure_rm_mariadbfirewallrule_info.py
./plugins/modules/azure_rm_mariadbserver_info.py
./plugins/modules/azure_rm_mysqlconfiguration_info.py
./plugins/modules/azure_rm_mysqldatabase_info.py
./plugins/modules/azure_rm_mysqlfirewallrule_info.py
./plugins/modules/azure_rm_mysqlserver_info.py
./plugins/modules/azure_rm_networkinterface_info.py
./plugins/modules/azure_rm_postgresqlconfiguration_info.py
./plugins/modules/azure_rm_postgresqldatabase_info.py
./plugins/modules/azure_rm_postgresqlfirewallrule_info.py
./plugins/modules/azure_rm_postgresqlserver_info.py
./plugins/modules/azure_rm_privatednsrecordset_info.py
./plugins/modules/azure_rm_privatednszone_info.py
./plugins/modules/azure_rm_publicipaddress_info.py
./plugins/modules/azure_rm_rediscache_info.py
./plugins/modules/azure_rm_registrationassignment_info.py
./plugins/modules/azure_rm_registrationdefinition_info.py
./plugins/modules/azure_rm_resourcegroup_info.py
./plugins/modules/azure_rm_resource_info.py
./plugins/modules/azure_rm_roleassignment_info.py
./plugins/modules/azure_rm_roledefinition_info.py
./plugins/modules/azure_rm_routetable_info.py
./plugins/modules/azure_rm_route_info.py
./plugins/modules/azure_rm_securitygroup_info.py
./plugins/modules/azure_rm_servicebus_info.py
./plugins/modules/azure_rm_sqldatabase_info.py
./plugins/modules/azure_rm_sqlfirewallrule_info.py
./plugins/modules/azure_rm_sqlserver_info.py
./plugins/modules/azure_rm_storageaccount_info.py
./plugins/modules/azure_rm_subnet_info.py
./plugins/modules/azure_rm_subscription_info.py
./plugins/modules/azure_rm_trafficmanagerendpoint_info.py
./plugins/modules/azure_rm_trafficmanagerprofile_info.py
./plugins/modules/azure_rm_virtualmachineextension_info.py
./plugins/modules/azure_rm_virtualmachineimage_info.py
./plugins/modules/azure_rm_virtualmachinescalesetextension_info.py
./plugins/modules/azure_rm_virtualmachinescalesetinstance_info.py
./plugins/modules/azure_rm_virtualmachinescaleset_info.py
./plugins/modules/azure_rm_virtualmachine_info.py
./plugins/modules/azure_rm_virtualnetworkpeering_info.py
./plugins/modules/azure_rm_virtualnetwork_info.py
./plugins/modules/azure_rm_virtualwan_info.py
./plugins/modules/azure_rm_vpnsitelink_info.py
./plugins/modules/azure_rm_vpnsite_info.py
./plugins/modules/azure_rm_webapp_info.py
```

For completeness: list of **non-info** modules that **do not** support check mode:
```
./plugins/modules/azure_rm_adpassword.py
./plugins/modules/azure_rm_adserviceprincipal.py
./plugins/modules/azure_rm_cognitivesearch.py
./plugins/modules/azure_rm_datalakestore.py
./plugins/modules/azure_rm_deployment.py
./plugins/modules/azure_rm_resource.py
./plugins/modules/azure_rm_virtualmachineextension.py
./plugins/modules/azure_rm_virtualmachinescalesetextension.py
./plugins/modules/azure_rm_virtualmachinescalesetinstance.py
```

And a list of ***_info** modules that **do** support check mode:
```
./plugins/modules/azure_rm_backupazurevm_info.py
./plugins/modules/azure_rm_iotdevice_info.py
./plugins/modules/azure_rm_manageddisk_info.py
./plugins/modules/azure_rm_recoveryservicesvault_info.py
./plugins/modules/azure_rm_vmbackuppolicy_info.py
```

In general I would like to see all modules supporting check mode.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
multiple

##### ADDITIONAL INFORMATION
I think check mode support is mandatory for _info modules:
https://github.com/ansible/ansible/blame/devel/docs/docsite/rst/dev_guide/developing_modules_general.rst#L82

**Comment by Fred-sun** [comment]
@Klaas- Thanks for you to submit this issue! I will confirm this!

**Comment by Fred-sun** [comment]
@Klaas- The check_mode mode should appear in the related modules azure_rm_*.py. The modules for azure_rm_*_info.py are only fetching resources and do not change them, so they should not support this functionality and remove modules that should not be there. Thank you very much!

**Comment by Fred-sun** [comment]
fixes by #596
